cli/tools/coverage/mod.rs

```diff
@@ -194,41 +201,89 @@ pub struct CoverageReport {
   output: Option<PathBuf>,
 }
 
+struct GenerateCoverageReportOptions<'a> {
+  cli_options: &'a CliOptions,
+  script_module_specifier: Url,
+  script_media_type: MediaType,
+  script_coverage: &'a cdp::ScriptCoverage,
+  script_original_source: String,
+  script_runtime_source: String,
+  maybe_source_map: &'a Option<Vec<u8>>,
+  output: &'a Option<PathBuf>,
+}
+
 fn generate_coverage_report(
-  script_coverage: &cdp::ScriptCoverage,
-  script_source: String,
-  maybe_source_map: &Option<Vec<u8>>,
-  output: &Option<PathBuf>,
+  options: GenerateCoverageReportOptions,
 ) -> CoverageReport {
-  let maybe_source_map = maybe_source_map
+  let parsed_source = parse_program(
+    options.script_module_specifier,
+    options.script_media_type,
+    &options.script_original_source,
+  )
+  .expect("invalid source code");
```

Comment:
We shouldn't use `expect` here - we're gonna find all sorts of strange or broken files that are included by mistake - we should error out or print a warning in this case instead of panicking.